src/test/e2e/psp.feature.ts

```diff
@@ -0,0 +1,353 @@
+        airnodeWallet.address,
+        templateId2,
+        encodedBeaconUpdateSubscriptionConditions2,
+        airnodeWallet.address,
+        roles.sponsor.address
+      );
+  });
+
+  it('updates the beacon successfully', async () => {
+    jest
+      .spyOn(config, 'loadAirnodeConfig')
+      .mockImplementationOnce(() => airnodeConfig as any)
+      .mockImplementationOnce(() => airnodeConfig as any);
+    jest.spyOn(config, 'loadAirkeeperConfig').mockImplementationOnce(() => airkeeperConfig as any);
+    const res = await psp.handler();
+
+    const voidSigner = new hre.ethers.VoidSigner(hre.ethers.constants.AddressZero, provider);
+
+    // ETH price beacon
+    const beaconIdETH = await dapiServer.subscriptionIdToBeaconId(subscriptionIdETH);
+    const dapiServerResponseETH = await dapiServer.connect(voidSigner).readWithDataPointId(beaconIdETH);
+
+    // BTC price beacon
+    const beaconIdBTC = await dapiServer.subscriptionIdToBeaconId(subscriptionIdBTC);
+    const dapiServerResponseBTC = await dapiServer.connect(voidSigner).readWithDataPointId(beaconIdBTC);
+
+    expect(dapiServerResponseETH[0].toNumber()).toEqual(723.39202 * 1000000);
+    expect(dapiServerResponseBTC[0].toNumber()).toEqual(41091.12345 * 1000000);
+    expect(res).toEqual({
+      statusCode: 200,
+      body: JSON.stringify({ ok: true, data: { message: 'PSP beacon update execution has finished' } }),
+    });
+  });
+
+  it('updates the beacon successfully after retrying a failed api call', async () => {
+    jest
+      .spyOn(config, 'loadAirnodeConfig')
+      .mockImplementationOnce(() => airnodeConfig as any)
+      .mockImplementationOnce(() => airnodeConfig as any);
+    jest.spyOn(config, 'loadAirkeeperConfig').mockImplementationOnce(() => airkeeperConfig as any);
+
+    const callApiSpy = jest.spyOn(api, 'callApi');
+    callApiSpy.mockRejectedValueOnce(new Error('Api call failed'));
+
+    const res = await psp.handler();
+
+    const voidSigner = new hre.ethers.VoidSigner(hre.ethers.constants.AddressZero, provider);
+
+    // ETH price beacon
+    const beaconIdETH = await dapiServer.subscriptionIdToBeaconId(subscriptionIdETH);
+    const dapiServerResponseETH = await dapiServer.connect(voidSigner).readWithDataPointId(beaconIdETH);
+
+    // BTC price beacon
+    const beaconIdBTC = await dapiServer.subscriptionIdToBeaconId(subscriptionIdBTC);
+    const dapiServerResponseBTC = await dapiServer.connect(voidSigner).readWithDataPointId(beaconIdBTC);
+
+    expect(dapiServerResponseETH[0].toNumber()).toEqual(723.39202 * 1000000);
+    expect(dapiServerResponseBTC[0].toNumber()).toEqual(41091.12345 * 1000000);
+    expect(res).toEqual({
+      statusCode: 200,
+      body: JSON.stringify({ ok: true, data: { message: 'PSP beacon update execution has finished' } }),
+    });
+  });
+
+  it('updates the beacon successfully with one invalid provider present', async () => {
+    jest
+      .spyOn(config, 'loadAirnodeConfig')
+      .mockImplementation(
```

Comment:
To answer your question, it does work, so `mockImplementation` is setting the default value to be returned after any `mockImplementationOnce` cases are handled

But in this case it's not needed since we want the airnode config to always include the invalid provider, I'll change this to just have `mockImplementation` :+1: